pkgs/development/compilers/ponyc/default.nix

```
@@ -32,8 +32,8 @@ stdenv.mkDerivation (rec {
       googletest = fetchFromGitHub {
         owner = "google";
         repo = "googletest";
-        rev = "release-1.10.0";
-        sha256 = "1zbmab9295scgg4z2vclgfgjchfjailjnvzc6f5x9jvlsdi3dpwz";
+        rev = "release-1.11.0";
```

```suggestion
        rev = "release-1.11.0";
```
I lack context here, so please enlighten me.
`googletest` [recommends](https://github.com/google/googletest/) to stick to HEAD.
Is there a particular reason for this version?
Is there a reason `googletest` package is not part of nixpkgs?

---

The information you seek is in lib/CMakeLists.txt

https://github.com/ponylang/ponyc/blob/ffa177c524771d476d5486825f9d3fc63f90769f/lib/CMakeLists.txt#L43-L46

The values are hard-coded into the build-system.

---

Thank you for the review. I did try a number of times to use the version of googletest that's in nixpkgs and failed miserably.  It seems to be very tightly coupled and after fighting it for a few days I didn't consider it worth the effort and potentially diverge from the versions used by every other distribution build.

---

If you search in nixpkgs, we're far from the only package that does this directly too:

https://github.com/NixOS/nixpkgs/search?q=googletest&type=

---

You can use the hardcoded version, but then add a comment to point to source of package version. (As you just told me.)
Also, as the package has inner dependencies, it will require an update script to be upgrade-able by bot. So, it is desirable to test a general package because of this. I understand you have made an attempt, but if it is possible to package package in isolation, it makes the upgrade process easier (no need for special upgrade script).

---

> diverge from the versions used by every other distribution build.

At Nix, the philosophy is always use latest as long as is not breaking software.
Notice: Duplicating versions makes general upgrade harder, duplicates builds, etc.
Ideally we should avoid specific versions and prefer a general package whenever possible.
`googletest` package is not generally packaged at nixpkgs yet, and it is just being duplicated at specific version.
What I am trying to figure out is if `googletest` downstream packages really require a specific version or would a general version would work for all/most.

In case of a specific version, there is two requirements:
* Comment the source of version at project -> file -> line. (Mandatory)
* Customized update script (to enable bot upgrades). (Desirable)